clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/provider/agent/DcosInstanceCachingAgent.groovy

```diff
@@ -55,15 +57,17 @@ class DcosInstanceCachingAgent implements CachingAgent, AccountAware {
 
   @Override
   String getAgentType() {
-    "${accountName}/${DcosInstanceCachingAgent.simpleName}"
+    "${accountName}/${clusterName}/${DcosInstanceCachingAgent.simpleName}"
```

Comment:
That might reduce some overhead since we would reduce the number of instances to switch between but that generally increases code complexity and honestly I don't know if the gains we would get are worth it compared to trying to get the adjustment to just do a secret agent per cluster.